handler.go

```diff
@@ -194,5 +194,25 @@ func (pset pathConfigSet) find(path string) (pc *pathConfig, subpath string) {
 	if i > 0 && strings.HasPrefix(path, pset[i-1].path+"/") {
 		return &pset[i-1], path[len(pset[i-1].path)+1:]
 	}
-	return nil, ""
+
+	// Otherwise now look for the shortest prefix pathConfig
+	// For example:
+	//  * Given pathConfigs {"/y", "/example/helloworld", "/"}
+	//  * Query "/x" should return "/"
+	//  * Query "/example/helloworld/foo" should return "/example/helloworld"
+	shortestWeight := len(path)
+	var shortestPrefixConfig *pathConfig
+	for i, ps := range pset {
+		if len(ps.path) >= len(path) {
+			// We previously didn't find the path by search, so any
+			// route with equal or greater length is NOT a match.
+			continue
+		}
+		rest := strings.TrimPrefix(path, ps.path)
+		if len(rest) < shortestWeight {
```

Comment:
Since you're walking the list in ascending lexicographical order, I think this will get hit at most once, since you will stumble across the shortest one first.  (Although, as I note above, I think you want the longest, but the principle still applies.)

Reply:
>Since you're walking the list in ascending lexicographical order, I think this will get hit at most once
Not really, if we have ["/", "/abc/", "/abc/def/", "/xyz/"] for "/abc/foo"
We'll hit this the first time at:
i = 0: "/", which gives a length of 7 "abc/foo"
i = 1: "/abc/" which gives a length of 3 "foo"

And it is "shortest length after stripping the prefix"

Reply:
It finally clicked to me that I am using the wrong wording, after your request to use the proper subpath. I've gone with "shortest subpath" and that makes sense for the longest prefix but shortest subpath. PTAL and thank you!